### DISCUSSION TITLE
Alpine.start() and Rails Turbo - Alpine has already been initialized on this page

### DISCUSSION THREAD

**Discussion by leoplct** [alpinejs/alpine#3819] [1. Help] [answered]
I am using Rails with Turbo and I initialize Alpine in this way

```
import Alpine from 'alpinejs'
window.Alpine = Alpine
document.addEventListener("turbo:load", function() { 

    // Alpinejs
    window.Alpine.start();

})
```

The problem is that I got this warning in console

`Alpine Warning: Alpine has already been initialized on this page. Calling Alpine.start() more than once can cause problems.`

How can I check if Alpine has been already initialized before call Alpine.start() ?

**Comment by ekwoka** [comment]
Just only call Alpine.start on the page initial load.

**Reply by leoplct** [reply, marked as answer]
Do you mean this?


```
import Alpine from 'alpinejs'
window.Alpine = Alpine

document.addEventListener("DOMContentLoaded", function(){
    // Alpinejs Loaded only once
    window.Alpine.start();
})


document.addEventListener("turbo:load", function() { 
})
```

**Reply by ekwoka** [reply]
yes

**Reply by sveredyuk** [reply]
I have Rails 7 with Turbo 8, and now when I add Alpine like this, any link click leads to a full page reload instead of a partial body reload like Turbo does without Alpine. Do you have something like this?
